adb_shell/exceptions.py

```diff
@@ -115,11 +115,11 @@ class UsbReadFailedError(Exception):
 
     """
     def __init__(self, msg, usb_error):
-        super(UsbReadFailedError, self).__init__(msg)
+        super(UsbReadFailedError, self).__init__(msg, usb_error)
         self.usb_error = usb_error
 
     def __str__(self):
-        return '%s: %s' % (super(UsbReadFailedError, self).__str__(), str(self.usb_error))
+        return '%s: %s' % self.args
```

Comment:
What is `self.args`?

Reply:
As above, it's the stashed arguments tuple provided by the base `Exception` class. In this case it's `(msg, usb_error)` from the instantiation, making this pretty much equivalent to the old code since the `super().__str__()` call would stringify the exception message (IIUC equivalent to `self.args[0]` also) and we were manually stringifying `self.usb_error` and sticking it on the end.